File: rwh/ch03/BookStore.py
# with a class
class Book:
	def __init__(self, isbn, title, authors):
		self.isbn = isbn
		self.title = title
		self.authors = authors

	def __str__(self):
		return ' '.join(['Book', str(self.isbn), self.title, str(self.authors)])

book = Book(9780135072455, 'Algebra of Programming', ['Richard Bird', 'Oege de Moor'])
print(book)

# with a named tuple, which is closer to the haskell way
import collections

Book2 = collections.namedtuple('Book', ['isbn', 'title', 'authors'])
book2 = Book2(9780135072455, 'Algebra of Programming', ['Richard Bird', 'Oege de Moor'])
print(book2)

File: rwh/ch03/BookStore.scala
case class Book(val isbn: Long, val title: String, val authors: List[String])

val book = Book(9780135072455L, "Algebra of Programming", List("Richard Bird", "Oege de Moor"))
println(book)

// add book review
type CustomerID = Long
type ReviewBody = String

case class BookReview(val book: Book, val customerID: CustomerID, val review: ReviewBody)
val review = BookReview(book, 12345, "A nice book!")
println(review)